cmd/buildah/images.go

```diff
@@ -300,34 +306,63 @@ func outputImages(ctx context.Context, systemContext *types.SystemContext, store
 		}
 
 	outer:
-		for name, tags := range imageReposToMap(image.Names) {
-			for _, tag := range tags {
-				if !matchesReference(name+":"+tag, argName) {
+		for repo, tagsAndDigests := range imageReposToMap(image.Names) {
+			for _, tagOrDigest := range tagsAndDigests {
+				imageName := ""
+				imageDigest := ""
+				imageTag := ""
+				if repo != "" && tagOrDigest != "" {
+					if digest.Digest(tagOrDigest).Validate() == nil {
+						imageDigest = tagOrDigest
+						imageName = repo + "@" + imageDigest
+					} else {
+						imageTag = tagOrDigest
+						if len(image.Digest) > 0 {
+							imageDigest = image.Digest.String()
+						}
+						imageName = repo + ":" + imageTag
+					}
+				}
+
+				if !matchesReference(imageName, argName) {
 					continue
 				}
 				found = true
 
-				if !matchesFilter(ctx, store, image, name+":"+tag, filters) {
+				if !matchesFilter(ctx, store, image, imageName, filters) {
 					continue
 				}
+				imageDigests := make([]string, 0, len(image.Digests))
```

Comment:
(Absolutely non-blocking: `imageDigests` is invariant to `repo`/`tagsAndDigests`, and can be computed outside of the loop. OTOH that would naively cause the value to be computed for all non-matching images as well, so, maybe compute it here, but only if not computed already. Yet again, this is it’s cheap enough.)

Reply:
I didn't have a strong feeling about it either way.  Changing it.